weltyt.py

```diff
@@ -106,11 +113,16 @@ def buy_lands(lands):
                 f'SELECT price FROM lands WHERE class = "{chosen_class}";'
             ).fetchone()[0]
             new_money_amount = money_amount - lands_price
-            cur.execute(
-                f'UPDATE resources SET amount = {new_money_amount}  WHERE type = "Money";'
-            )
+            if new_money_amount < 0:
+                cur.execute("DELETE FROM lands WHERE id = (SELECT max(id) FROM lands)")
```

Comment:
Maybe you can check money before even inserting the new land? In that way additional delete will be unnecessary.

Reply:
Before inserting we don't know, how much land costs. We have to compare how much money we have, and what we can buy. I think that this is the easiest way.